code/modules/mob/mob.dm

```diff
@@ -114,9 +114,19 @@
 	//Process text
 	if(text)
 		to_chat(src, "<span class='italics'>[text]</span>")
-	//to whoever sees this: icons are handled in living.dm
-
-
+	//Process icon
+	if(sound_ring && audiolocation)
```

Comment:
Remove the audiolocation var on the mob and make the component handle the different behaviour by using a new signal added to this proc